### PR TITLE
Fix Class cast exception try to get clone readonly type with enum

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/32083

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BType.java [hunk 1]
```diff
@@ -190,4 +190,8 @@ private boolean hasAllNullConstituents(Module module) {
     public Module getPkg() {
         return pkg;
     }
+
+    public long getFlags(){
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BType.java:194]:
```suggestion
    public long getFlags() {
```

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java [hunk 2]
```diff
@@ -132,7 +132,8 @@ private static Type getAvailableImmutableType(Type type) {
 
 
     private static Type getImmutableType(Type type, Set<Type> unresolvedTypes) {
-        if (TypeChecker.isInherentlyImmutableType(type)) {
+        if (TypeChecker.isInherentlyImmutableType(type) || (SymbolFlags.isFlagOn(type.getFlags(),
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java:135]: Wondering if https://github.com/ballerina-platform/ballerina-lang/blob/80e4401bd9c0ce1f0059eee3111954d74d6d78b2/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java#L116 also needs to be updated. I'm not sure, may have to check that path.

**Review by MaryamZi — DISMISSED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java (hunk 2 above)
Comment (MaryamZi) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java:135]: Shall we create an issue? Error unions are also effectively immutable. We need to check and sync the behaviour with compile-time behaviour if there are any differences, IMO.

**Review by MaryamZi — DISMISSED** [review]

**Review by MaryamZi — APPROVED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java (hunk 2 above)
Comment (warunalakshitha) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java:135]: Added https://github.com/ballerina-platform/ballerina-lang/issues/32083

**Pull request merged**